src/nifreeze/utils/iterators.py

```diff
@@ -53,23 +54,24 @@ def linear_iterator(size: int = None, **kwargs) -> Iterator[int]:
     if size is None:
         raise TypeError("Cannot build iterator without size")
 
-    return range(size)
+    return iter(range(size))
 
 
-def random_iterator(size: int = None, **kwargs) -> Iterator[int]:
+def random_iterator(size: int | None = None, **kwargs) -> Iterator[int]:
     """
     Traverse the dataset volumes randomly.
 
+    If the `seed` key is present in the keyword arguments, initializes the seed
+    of Python's `random` pseudo-random number generator library with the given
+    value. Specifically, if `False`, `None` is used as the seed; it `True`, a
+    default seed value is used.
+
     Parameters
     ----------
-    size : :obj:`int`
-        Number of volumes in the dataset
-        (for instance, the number of orientations in a DWI)
-    seed : :obj:`int` or :obj:`bool` or :obj:`bool` or ``None``
-        If :obj:`int` or :obj:`str` or ``None``, initializes the seed of Python's random generator
-        with the given value.
-        If ``False``, the random generator is passed ``None``.
-        If ``True``, a default seed value is set.
+    size : :obj:`int` or None, optional
+        Number of volumes in the dataset (for instance, the number of
+        orientations in a DWI). If `None`, the size is inferred from the `bvals`
+        argument in `kwargs`, if present.
```

Comment:
Although it is not explicitly stated in the signature, I think we can use the opportunity that sphinx allows us to document seed here if we want. Removing the specification of seed here makes this documentation less accessible and useful.
```suggestion
    size : :obj:`int` or None, optional
        Number of volumes in the dataset (for instance, the number of
        orientations in a DWI). If `None`, the size is inferred from the `bvals`
        argument in `kwargs`, if present.
    seed : :obj:`int` or :obj:`bool` or :obj:`bool` or ``None``
        If :obj:`int` or :obj:`str` or ``None``, initializes the seed of Python's random generator
        with the given value.
        If ``False``, the random generator is passed ``None``.
        If ``True``, a default seed value is set.
```

Reply:
As per numpydoc guidelines, [this seems to be a use case for *Other parameters*](https://numpydoc.readthedocs.io/en/latest/format.html#other-parameters). Updated the suggestion.

Reply:
That said, I don't think it clutters the parameters section in this case, so I would just leave it with the parameters section.